server/py/framework/db/sqldb/db.py

```diff
@@ -3185,6 +3203,88 @@ def _calculate_runs_counters(
             project_to_running_runs_count,
         )
 
+    def _calculate_alert_activations_counters(
+        self,
+        session,
+        projects_with_creation_time: list[tuple[str, datetime]],
+    ) -> tuple[
+        dict[str, int],
+        dict[str, int],
+        dict[str, int],
+    ]:
+        project_to_endpoint_alerts_count = collections.defaultdict(int)
+        project_to_job_alerts_count = collections.defaultdict(int)
+        project_to_other_alerts_count = collections.defaultdict(int)
+
+        project_filter_conditions = self._generate_project_filter_conditions(
+            projects_with_creation_time
+        )
+
+        last_day = mlrun.utils.datetime_now() - timedelta(hours=24)
+
+        query_results = (
+            session.query(
+                AlertActivation.project,
+                func.count(
+                    case(
+                        (
+                            AlertActivation.entity_kind
+                            == mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
+                            1,
+                        )
+                    )
+                ).label("model_endpoint_alerts_count"),
+                func.count(
+                    case(
+                        (
+                            AlertActivation.entity_kind
+                            == mlrun.common.schemas.alert.EventEntityKind.JOB,
+                            1,
+                        )
+                    )
+                ).label("job_alerts_count"),
+                func.count(
+                    case(
+                        (
+                            ~AlertActivation.entity_kind.in_(
+                                [
+                                    mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
+                                    mlrun.common.schemas.alert.EventEntityKind.JOB,
+                                ]
+                            ),
+                            1,
+                        )
+                    )
+                ).label("other_alerts_count"),
+            )
+            .filter(or_(*project_filter_conditions))
+            .filter(AlertActivation.activation_time > last_day)
+            .group_by(AlertActivation.project)
+            .all()
+        )
+        for project, endpoint_counter, job_counter, other_counter in query_results:
+            project_to_endpoint_alerts_count[project] = endpoint_counter
+            project_to_job_alerts_count[project] = job_counter
+            project_to_other_alerts_count[project] = other_counter
+
+        return (
+            project_to_endpoint_alerts_count,
+            project_to_job_alerts_count,
+            project_to_other_alerts_count,
+        )
+
+    @staticmethod
+    def _generate_project_filter_conditions(
```

Comment:
this is alert-activiation specific implementation, yet its name is very general
I suggest to move it to where it is being used or choose a proper function name